pandas/tests/indexes/test_interval.py

```diff
@@ -395,25 +597,129 @@ def slice_locs_decreasing_cases(self, tuples):
-        pytest.raises(KeyError, self.index.get_loc, Interval(2, 3))
-        pytest.raises(KeyError, self.index.get_loc,
-                      Interval(-1, 0, 'left'))
+    ### NEW SLICE LOCS BEHAVIOR
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def slice_locs_with_interval_updated_behavior(self):
+        # @jreback might prefer these be broken up into separate methods
+
+        # increasing
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == (0, 1)
+        assert index.slice_locs(start=Interval(0, 2)) == (0, 1, 2)
+        assert index.slice_locs(end=Interval(2, 4)) == (0, 1)
+        assert index.slice_locs(end=Interval(0, 2)) == ()
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == ()
+
+        # decreasing
+        index = IntervalIndex.from_tuples([(2, 4), (1, 3), (0, 2)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == ()
+        assert index.slice_locs(start=Interval(0, 2)) == (2)
+        assert index.slice_locs(end=Interval(2, 4)) == ()
+        assert index.slice_locs(end=Interval(0, 2)) == (0, 1)
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == (0, 1)
+
+        # sorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == # ?
+        assert index.slice_locs(start=Interval(0, 2)) == # ?
+        assert index.slice_locs(end=Interval(2, 4)) == # ?
+        assert index.slice_locs(end=Interval(0, 2)) == # ?
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == # ?
+
+        # unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (2, 4), (0, 2)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == # ?
+        assert index.slice_locs(start=Interval(0, 2)) == # ?
+        assert index.slice_locs(end=Interval(2, 4)) == # ?
+        assert index.slice_locs(end=Interval(0, 2)) == # ?
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == # ?
+
+        # different unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4), (1, 3)])
+
+        assert index.slice_locs(start=Interval(0, 2), end=Interval(2, 4)) == # ?
+        assert index.slice_locs(start=Interval(0, 2)) == # ?
+        assert index.slice_locs(end=Interval(2, 4)) == # ?
+        assert index.slice_locs(end=Interval(0, 2)) == # ?
+        assert index.slice_locs(start=Interval(2, 4), end=Interval(0, 2)) == # ?
+
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def slice_locs_with_ints_and_floats_updated_behavior(self):
+
+        # increasing
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+
+        assert index.slice_locs(0, 1) == #?
```

Comment:
This one I am not sure about what to do with it ...

Possibilities (considering the Interval(start, end) for the slice bounds):

- select all intervals that are strictly contained by the Interval(start, end)
- select all intervals that at least overlap with Interval(start, end)
- raise an error for now (keep slicing for exact Interval matches, and achieve this kind of indexing with booleans indexing using the new `overlaps`, `contains` methods)
- ...

First two of course would only work for sorted index